go/genesis/grpc.go

```diff
@@ -72,18 +74,26 @@ func (s *grpcServer) ToGenesis(ctx context.Context, req *pb.GenesisRequest) (*pb
 	if err != nil {
 		return nil, err
 	}
+	schedulerGenesis, err := s.schedulerBackend.ToGenesis(ctx, height)
+	if err != nil {
+		return nil, err
+	}
 
 	doc := api.Document{
 		// XXX: Tendermint doesn't support restoring from non-0 height.
 		// https://github.com/tendermint/tendermint/issues/2543
-		Height:     0,
-		ChainID:    genesisDoc.ChainID,
-		Time:       time.Now(),
-		EpochTime:  *epochtimeGenesis,
-		Registry:   *registryGenesis,
-		RootHash:   *roothashGenesis,
-		Staking:    *stakingGenesis,
-		KeyManager: *keymanagerGenesis,
+		Height:           0,
```

Comment:
Unrelated to your PR, we should record the height at which the dump was taken and just log/ignore it when restoring.